pkg/common/cache.go

```diff
@@ -0,0 +1,62 @@
+/*
```

Comment:
nit: Should we name this file `handler_cache` to be more descriptive and in line with the structs?

Reply:
Makes sense, fixed!